### DISCUSSION TITLE
How to implement Angular Dynamic routing on Github Pages

### DISCUSSION THREAD

**Discussion by Gabri432** [community/community#66701] [Other Feature Feedback, Questions, & Ideas]
### Body

I have published a [personal blog website](https://github.com/Gabri432/angular-personal-website) made with Angular (16+) and **hosted on Github Pages**.

I have followed [this instructions](https://angular.io/guide/deployment#deploy-to-github-pages) on how to deploy an Angular app on github pages. I have copied the content of index.html in 404.html as said.

All the content is displayed and every route works as expected, but Github Pages will still issue a 404 error for every page that is not the main one.

Because of this **I cannot use Google Chrome Lighthouse** to test for SEO and performances. Also it means that those other pages won't be indexed by Google.

I have read the solution given in [this discussion](https://github.com/orgs/community/discussions/22901), but it is from 2018 and maybe some other solutions have been found.

Another solution would be to implement hashed routing, but this option may hurt SEO.

Is there a solution to my problem? Specifically, is there a way to avoid getting 404 from Github Pages? Thanks in advance.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by yeabtsegataye** [comment]
try chat gpt

**Reply by Gabri432** [reply]
I have actually asked to chatGPT, which said to add a sitemap.xml and a robots.txt to allow google to index my pages despite being in a 404.html.

Google did find and read my sitemap.xml, but it still cannot find the content because Github Pages keep returning 404 for all dynamic pages.

**Comment by GhostOf0days** [comment]
Github Pages can only serve static websites. If you want a dynamic website (I would actually not ever use Github Pages), use Netlify, Heroku, Vercel, or Railway.

**Comment by Gabri432** [comment]
At the end I have accomplished what I was trying to do. 

I have discovered and used Scully to prerender my Angular components and it worked as expected. Now Google can crawl all the pages of my website.

**Reply by bigezoacademy** [reply]
Thanks for suggesting this solution. let me try it out because i am facing the same issue

**Reply by IPdotSetAF** [reply]
As Scully is not actively maintained anymore, A Better option is to configure your angular project for SSG.
